dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/AggregationLevelsHelper.java

```diff
@@ -0,0 +1,272 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.table;
+
+import static org.hisp.dhis.commons.util.TextUtils.replace;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.hisp.dhis.analytics.DataQueryParams;
+import org.hisp.dhis.common.IllegalQueryException;
+import org.hisp.dhis.db.model.Column;
+import org.hisp.dhis.db.model.Table;
+import org.hisp.dhis.db.sql.SqlBuilder;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+/**
+ * Helper class for applying aggregation level updates to analytics tables in both Apache Doris and
+ * PostgreSQL.
+ *
+ * <p>This class provides utilities to update or nullify organization unit level columns (e.g.,
+ * level1, level2, ...) in partitioned or non-partitioned analytics tables, using the appropriate
+ * strategy for the underlying database:
+ *
+ * <ul>
+ *   <li><b>Doris:</b> Uses an "Insert and Delete" strategy, which involves a staging table and
+ *       partition-aware deletes.
+ *   <li><b>PostgreSQL:</b> Uses a direct UPDATE SQL statement to set affected columns to NULL.
+ * </ul>
+ */
+@Slf4j
+public class AggregationLevelsHelper {
+
+  private final JdbcTemplate jdbcTemplate;
+  private final SqlBuilder sqlBuilder;
+
+  public AggregationLevelsHelper(JdbcTemplate jdbcTemplate, SqlBuilder sqlBuilder) {
+    this.jdbcTemplate = jdbcTemplate;
+    this.sqlBuilder = sqlBuilder;
+  }
+
+  private String quote(String identifier) {
+    return sqlBuilder.quote(identifier);
+  }
+
+  /** Returns a comma-separated, single-quoted string list for SQL IN clause. */
+  private String quotedCommaDelimitedString(Collection<String> values) {
+    return values.stream()
+        .map(v -> "'" + v.replace("'", "''") + "'")
+        .collect(Collectors.joining(", "));
+  }
+
+  /** Fetches all partition names from a Doris table, regardless of partition column. */
+  private List<String> getAllPartitions(String tableName) {
+    return jdbcTemplate.query(
+        "show partitions from " + quote(tableName), (rs, rowNum) -> rs.getString("PartitionName"));
+  }
+
+  /** Fetches all column names from the target table. */
+  private List<String> getColumnNames(Table table) {
+
+    if (table.getColumns().isEmpty()) {
+      Table parent = table.getParent();
+      if (parent != null) {
+        return parent.getColumns().stream().map(Column::getName).toList();
+      }
+    } else {
+      return table.getColumns().stream().map(Column::getName).toList();
+    }
+    throw new IllegalQueryException("No columns found for table: " + table.getName());
```

Comment:
If `(parent == null)` we will throw `No columns found for table`.
Should we throw something like `No parent table found for TABLE_NAME`?